File: exam/string/palindrome/palindrome.dart
void main() {
  String data = "MOM";
  String data2 = "aaabbaaa";
  print(isPalindrome(data));

  print(isPalindrome(data2));

  print("Result : ${leastString(data)}");
  print("Result 2: ${leastString(data2)}");
}

String leastString(String data) {
  if (!isPalindrome(data)) {
    return 'Not Possible';
  } else {
    List<String> res = data.split("").toList();
    for (var i = 0; i < res.length; i++) {
      if (res[i] != 'a') {
        res[i] = 'a';
        String result = res.map((e) => e).join();
        return result;
      }
    }
    return "Not Possible";
  }
}

bool isPalindrome(String data) {
  String newSet = data.split("").toList().reversed.join();

  if (data == newSet) {
    return true;
  } else {
    return false;
  }
}

File: exam/string/palindrome/palindrome.cpp
#include <iostream>
using namespace std;

bool isPalindrome(string dara)
{
    int start = 0;
    int end = dara.length() - 1;

    if (dara[start] != dara[end])
    {
        return false;
    }
    else
    {
        return true;
    }
}

void isLeastNewPalindrome(string res)
{
    if (!isPalindrome(res))
    {
        cout << "Not Possible " << endl;
    }
    else
    {
        for (int i = 0; i < res.length() - 1; i++)
        {
            if (res[i] != 'a')
            {
                res[i] = 'a';
                cout << res << endl;
                break;
            }
        }
        cout << "Not Possible " << endl;
    }
}

int main()
{
    string data = "MOM";
    string data1 = "MOMA";
    string data2 = "aaabbaaa";
    cout << isPalindrome(data) << endl;
    cout << isPalindrome(data1) << endl;
    isLeastNewPalindrome(data);
    isLeastNewPalindrome(data1);
    isLeastNewPalindrome(data2);
}